### PR TITLE
feat(core): add syntax highlighting and edit button

### PR DESCRIPTION
**Pull request by oysteinsigholt**

### CODE REVIEW

**Comment by oysteinsigholt** [comment]
Ignore travis, as it hasn't been properly set up yet

File: lib/postpluginprocessor.js [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
+
+// give each file a url field
+function addUrl(files, metalsmith) {
+    Object.keys(files).forEach((file) => {
+        files[file].url = config.webRoot + file;
+    });
+}
+
+// remove files that haven't been processed from navigation
+function removeNavigation(files, metalsmith) {
```
Comment (jorunfa) [inline lib/postpluginprocessor.js]: Rename to `removeUnprocessedFilesFromNavigation` instead of comment?

File: lib/postpluginprocessor.js [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
-            if (!(/\.html$/.test(path.extname(file)))) delete files[file].nav_groups;
-        });
+        addUrl(files, metalsmith);
+        removeNavigation(files, metalsmith);
         done();
     };
 };
+
+// give each file a url field
+function addUrl(files, metalsmith) {
```
Comment (jorunfa) [inline lib/postpluginprocessor.js]: Remove comment? Why do we pass metalsmith if it's not used?

File: lib/postpluginprocessor.js (hunk 1 above)
Comment (jorunfa) [inline lib/postpluginprocessor.js]: Why do we pass metalsmith if it's not used?

File: lib/prepluginprocessor.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+module.exports = () => {
+    return (files, metalsmith, done) => {
+        addEditUrl(files, metalsmith);
+        done();
+    };
+};
+
+// give each file an editUrl field if config.editUrl exists
+function addEditUrl(files, metalsmith) {
```
Comment (jorunfa) [inline lib/prepluginprocessor.js]: Remove comment and change to `addEditUrlIfExists`? Remove `metalsmith` param?

**Comment by jorunfa** [comment]
LGTM

**Pull request merged**